### PR TITLE
Support metering cost for memory pages

### PR DESCRIPTION
**Pull request by axic**
Addresses https://github.com/wasmx/fizzy/issues/621#issuecomment-1298594860.

### CODE REVIEW

File: lib/fizzy/instructions.hpp [hunk 1]
```diff
@@ -31,4 +31,10 @@ const uint8_t* get_instruction_max_align_table() noexcept;
 /// execution metering.
 const int16_t* get_instruction_cost_table() noexcept;
 
+/// Calculates the cost of of memory expansion. Currently set at 65536 per page (1 per byte).
```
Comment (axic) [inline lib/fizzy/instructions.hpp:34]:
```suggestion
/// Calculates the cost of memory expansion. Currently set at 65536 per page (1 per byte).
```

**Review by chfast — APPROVED** [review]

**Comment by axic** [comment]
@gumb0 any comments here?

File: lib/fizzy/execute.cpp [hunk 2]
```diff
@@ -898,6 +898,12 @@ ExecutionResult execute(
         }
         case Instr::memory_grow:
         {
+            if constexpr (MeteringEnabled)
+            {
+                if ((ctx.ticks -= get_grow_memory_cost(stack.top().as<uint32_t>())) < 0)
```
Comment (gumb0) [inline lib/fizzy/execute.cpp:903]: I would maybe define `const auto delta = stack.top().as<uint32_t>();` to use here and below

**Review by gumb0 — APPROVED** [review]

File: lib/fizzy/execute.cpp (hunk 2 above)
Comment (axic) [reply lib/fizzy/execute.cpp:903]: Using `delta_pages` as that is used in both `grow_memory` and `get_grow_memory_cost`.

**Pull request merged**